storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -81,6 +84,14 @@ public void addOrUpdate(Storable storable) throws StorageException {
     }
 
     @Override
+    public void update(Storable storable) {
+        if (queryExecutor.update(storable) == 0) {
```

Comment:
I am not quite following the relation between updating 0 rows and concurrent update? Shouldn't the ConcurrentUpdateException be thrown in queryExecutor.update(storable) method instead?